Formula/zsh-fast-syntax-highlighting.rb

```diff
@@ -0,0 +1,33 @@
+class ZshFastSyntaxHighlighting < Formula
+  desc "Feature-rich syntax highlighting for Zsh"
+  homepage "https://github.com/zdharma-continuum/fast-syntax-highlighting"
+  url "https://github.com/zdharma-continuum/fast-syntax-highlighting/archive/refs/tags/v1.55.tar.gz"
+  sha256 "d06cea9c047ce46ad09ffd01a8489a849fc65b8b6310bd08f8bcec9d6f81a898"
+  license "BSD-3-Clause"
+  head "https://github.com/zdharma-continuum/fast-syntax-highlighting.git", branch: "master"
+
+  uses_from_macos "zsh" => [:build, :test]
+
+  def install
+    pkgshare.install Dir["*"]
+  end
+
+  def caveats
+    <<~EOS
+      To activate the syntax highlighting, add the following at the end of your .zshrc:
+        source #{opt_pkgshare}/fast-syntax-highlighting.plugin.zsh
+    EOS
+  end
+
+  test do
+    test_script = testpath/"script.zsh"
+    test_script.write <<~ZSH
+      #!/usr/bin/env zsh
+      source #{pkgshare}/fast-syntax-highlighting.plugin.zsh
+      printf '%s' ${FAST_HIGHLIGHT_STYLES+yes}
+    ZSH
+    test_out, test_err, test_ret = Open3.capture3('zsh', test_script)
+    assert_equal 0, test_ret
+    assert_match "yes", test_out
```

Comment:
```suggestion
    assert_match "yes", shell_output("zsh #{test_script}")
```
Does this not work?

Reply:
The Ruby API doc (https://rubydoc.brew.sh/Homebrew/Assertions.html#shell_output-instance_method) doesn't explain what `shell_output` accepts as input. Does it just accept a string that is run as a shell script?

Reply:
Wouldn't it be safer to have an option that accepts a command + arguments instead of a shell script? And also be able to check exit statuses and stderr as well as stdout?

Reply:
> Wouldn't it be safer to have an option that accepts a command + arguments instead of a shell script? And also be able to check exit statuses and stderr as well as stdout?

It can do all of the above, for example: `shell_output("#{bin}/fail_cmd some --args 2>&1", 255)`.
- arguments speak for themselves
- `2>&1` redirects stderr so you can check it
- `255` is the expected exit code.